src/Servers/HttpSys/src/RequestProcessing/RequestContext.FeatureCollection.cs

```diff
@@ -728,5 +739,14 @@ public void DelegateRequest(DelegationRule destination)
             Delegate(destination);
             _responseStarted = true;
         }
+
+        void IConnectionLifetimeNotificationFeature.RequestClose()
+        {
+            // Set the connection close feature if the response hasn't sent headers as yet
+            if (!Response.HasStarted)
+            {
+                Response.Headers[HeaderNames.Connection] = "close";
```

Comment:
FYI the actual close logic is over here:
https://github.com/dotnet/aspnetcore/blob/4c9c303b971b99169a14b42a35d95eb47e67ebfb/src/Servers/HttpSys/src/RequestProcessing/Response.cs#L464-L477

Setting the close header is not required for the HTTP/2 code path, it gets converted to a flag instead, but this is adequate for now.